### ISSUE TITLE
Problematic parameter in RSASA asarray()

### ISSUE THREAD

**Issue by jchutrue** [nelpy/nelpy#284]
On the develop branch, in `asarray()`, the docstring says 
https://github.com/nelpy/nelpy/blob/c0ad713f39b1dc71b25fb6727aae831f945811f5/nelpy/core/_analogsignalarray.py#L1711-L1730

If both `where` and `at` are `None` and `n_samples` are specified, then this block of code gets executed. 
https://github.com/nelpy/nelpy/blob/c0ad713f39b1dc71b25fb6727aae831f945811f5/nelpy/core/_analogsignalarray.py#L1762-L1763

If the support consists of multiple intervals, the `n_samples` parameter is problematic for two reasons:
1. Before we started dropping points that lie outside the support (see #160), sampling uniformly from `self.support.start` to `self.support.stop` and then interpolating could result in NaNs, if any of the points were not in the support.
2. Now that we drop points that lie outside the support, we can no longer guarantee that interpolated samples are uniformly spaced.

**Comment by eackermann** [comment]
Good catch. The machinery to get `n_samples` within the supports is already being used in `asa.simplify()`, but we should probably refactor that machinery out, so that we can use it in `simplify`, or in `asarray`, or wherever. It is also not restricted to ASAs, since a method that returns `n_samples` uniformly spaced in an interval array can be generally useful.

It might even make sense to make it an `IntervalArray` method instead? Not sure though. We do have `IntervalArray.partition(n_intervals)`, which is similar in concept. What do you suggest, @jchutrue ?

At a minimum, it should be refactored out of simplify, and used in both simplify and asarray. But perhaps we should elevate it's scope to make it accessible to any object that may want samples distributed in it's support.
